### ISSUE TITLE
[Test] topk_softmax input_dtype parameterization always exercises FP32 inputs

### ISSUE THREAD

**Issue by lukatao** [flagos-ai/FlagGems#6382]
### Describe the bug

`test_topk_softmax` parameterizes `input_dtype` over FP32, FP16, and BF16, but creates `gating_output` with a hard-coded `dtype=torch.float32`. The `input_dtype` argument is unused, and both the Gems and vLLM calls receive FP32 inputs in every case.

Consequently, the FP16/BF16 cases repeat FP32 coverage instead of exercising the corresponding Gems input paths. This is a test coverage issue; it does not establish a defect in the operator implementation.

Affected code on master at `77711d9d055d6d9484af9c13f5ede61aed9c2c5c`:
- [Input dtype parameterization](https://github.com/flagos-ai/FlagGems/blob/77711d9d055d6d9484af9c13f5ede61aed9c2c5c/tests/test_topk_softmax.py#L56-L60)
- [Hard-coded FP32 input and the two calls](https://github.com/flagos-ai/FlagGems/blob/77711d9d055d6d9484af9c13f5ede61aed9c2c5c/tests/test_topk_softmax.py#L73-L99)

#### Minimal illustration of the input-generation problem

```python
import torch

for input_dtype in (torch.float32, torch.float16, torch.bfloat16):
    # Same dtype selection as the test; device is immaterial here.
    gating_output = torch.randn(4, 8, dtype=torch.float32)
    print(input_dtype, gating_output.dtype)
```

All three iterations produce FP32 tensors, including the cases labeled FP16 and BF16.

#### Expected behavior / possible fix

Generate inputs using `dtype=input_dtype` and pass those tensors unchanged to Gems. A check such as `assert gating_output.dtype == input_dtype` can make the intended coverage explicit.

The reference call should account for the vLLM version used by the test environment. vLLM's older [v0.10.2 implementation](https://github.com/vllm-project/vllm/blob/v0.10.2/csrc/moe/topk_softmax_kernels.cu#L486-L539) reads FP32 logits, while its [current main implementation](https://github.com/vllm-project/vllm/blob/main/csrc/libtorch_stable/moe/topk_softmax_kernels.cu#L760-L794) dispatches FP32, FP16, and BF16 inputs. It would therefore be inaccurate to assume that vLLM universally requires FP32 logits.

For a compatible vLLM version supporting these input dtypes, both calls can receive the same tensor directly. If the supported reference version requires FP32 logits, only the reference input can be promoted with `gating_output.float()`, preserving the values already rounded to the tested input dtype while Gems still receives FP16/BF16. The reference API signature and numerical comparisons should be verified against the supported version.
